### ISSUE TITLE
var vs let vs const

### ISSUE THREAD

**Issue by serge-nikitin** [jscraft-org/mentoring-project-2#11]
I don't know any good reasons to use `var`  (if we can use babel or modern JS engines).

By default we should always use `const`. We use `let` only if we really reassign variable.

In Skype I'll tell you more why it's a big deal.

**Comment by odv** [comment]
I did some research today on why I wouldn't want to use var.

To be honest, I didn't pay attention to code quality and was thinking that as long as something works, no need to think about what implications could a certain piece of code have or not and never had intentions to factor it.

One important lesson that I've learned now from you is to always think about that and always think about how to refactor your code.

So to get back to the helpful observation that you made, I learned the following:

**var** variables are function scoped and they can actually produce leaks, like here:

```
var name = 'Ovidiu';

if (name === 'Ovidiu') {
    var lastName = 'Voicu'
}
console.log(lastName);
```
lastName leaks outside of the if statement

Also learned from you that variables defined with **const** are not re-assignable while those defined with **let** are.
